src/sandstorm/run-bundle.c++

```diff
@@ -205,6 +205,42 @@ struct UserIds {
   kj::Array<gid_t> groups;
 };
 
+kj::Maybe<kj::Array<uint>> getPorts(uint httpsPort, kj::StringPtr portList) {
+    // TODO: Whose responsibility is it to bail out if httpsPort shows up
+    // in portList? Presumably I don't want to try binding on the same
+    // port multiple times down the road. I can live with deciding that's
+    // a semantics error (dealt with when it's time to bind to things)
+    // rather than a syntax error (dealt with now), but I noticed that I
+    // actually INTRODUCED this problem on my dev setup because I was
+    // careless, so presumably other people will be equally careless.8
+    auto portsSplitOnComma = split(portList, ',');
+    size_t arraySize = portsSplitOnComma.size();
+    if (httpsPort) {
+        arraySize += 1;
+    }
+    if (arraySize == 0) {
+        return nullptr;
+    }
+    auto result = kj::heapArray<uint>(arraySize);
+
+    // If there a HTTPS port, inject it into the array first.
+    size_t i = 0;
+    if (httpsPort) {
+        result[i] = httpsPort;
+        i += 1;
+    }
+    for (; i < portsSplitOnComma.size(); i++) {
+      KJ_IF_MAYBE(portNumber, parseUInt(kj::heapString(portsSplitOnComma[i]), 10)) {
```

Comment:
Replace `kj::heapString` with `trim` -- this helper will remove leading and trailing whitespace and will also conveniently accept an `ArrayPtr<const char>` and return a `String`.